### PR TITLE
feat(stats-detectors): Auto resolve stats detectors

### PR DESCRIPTION
**Pull request by Zylphrex**
When the long moving average approaches within 10% of the original baseline, auto resolve the issue.

### CODE REVIEW

File: tests/sentry/statistical_detectors/test_algorithm.py [hunk 1]
```diff
@@ -186,6 +186,26 @@ def test_moving_average_detector_state_from_redis_dict_error(data, error):
         MovingAverageDetectorState.from_redis_dict(data)
 
 
+@pytest.mark.parametrize(
+    ["baseline", "rel_threshold", "auto_resolve"],
+    [
+        pytest.param(100, 0.1, True, id="equal"),
+        pytest.param(105, 0.1, True, id="within threshold above"),
+        pytest.param(95, 0.1, True, id="within threshold below"),
+        pytest.param(115, 0.1, False, id="exceed threshold above"),
```
Comment (narsaynorath) [inline tests/sentry/statistical_detectors/test_algorithm.py:195]: With this test, are we saying that if the EMA drops below the relative threshold factor, we won't mark it as auto-resolved? It's only resolved if the range that the new behaviour is exhibiting is within the baseline +/- the threshold around it?

I imagine that dropping well below would be a potential auto-resolution event

File: tests/sentry/tasks/test_statistical_detectors.py [hunk 2]
```diff
@@ -571,6 +631,66 @@ def test_detect_function_trends(
...
+            active=True,
+            project_id=project.id,
+            fingerprint=f"{123:x}",
+            baseline=100,
+            regressed=300,
+        )
+        for ts in timestamps[20:]:
+            detect_function_trends([project.id], ts)
+
+    assert produce_occurrence_to_kafka.called
```
Comment (narsaynorath) [inline tests/sentry/tasks/test_statistical_detectors.py:691]: If we're going so far as to check that the occurrence is sent to kafka, should we also just add a check to know the correct status change is occurring?

**Review by narsaynorath — COMMENTED** [review]
just some small questions

**Review by narsaynorath — APPROVED** [review]

File: tests/sentry/statistical_detectors/test_algorithm.py (hunk 1 above)
Comment (Zylphrex) [reply tests/sentry/statistical_detectors/test_algorithm.py:195]: That's a good point, if it ever dips below 1.1x time the baseline, I think that's a reason to resolve it. Let me fix that up

**Pull request merged**